File: test/solutions/14.jl
function longestCommonPrefix(strs::Array{String,1})::AbstractString
    n = length(strs)
    if n == 0
        return ""
    elseif n == 1
        return strs[1]
    end
    min_len = minimum(length.(strs))
    common_prefix = ""
    for i in 1:min_len
        char_to_match = strs[1][i]
        for j in 2:n
            if strs[j][i] != char_to_match
                return common_prefix
            end
        end
        common_prefix *= char_to_match
    end
    return common_prefix
end


@testset "longestCommonPrefix tests" begin
    # Test case 1
    strs = ["flower", "flow", "flight"]
    expected_output = "fl"
    @test longestCommonPrefix(strs) == expected_output
    
    # Test case 2
    strs = ["dog", "racecar", "car"]
    expected_output = ""
    @test longestCommonPrefix(strs) == expected_output
    
    # Test case 3
    strs = ["test", "testing", "testable"]
    expected_output = "test"
    @test longestCommonPrefix(strs) == expected_output
end

File: test/rust/tests/14.rs
fn longest_common_prefix(strs: Vec<String>) -> String {
    if strs.is_empty() {
        return "".to_string();
    }

    let mut prefix = &strs[0];

    for s in &strs[1..] {
        while !s.starts_with(prefix) {
            prefix = &prefix[..prefix.len() - 1];
            if prefix.is_empty() {
                return "".to_string();
            }
        }
    }

    prefix.to_string()
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn test_example1() {
        let strs = vec!["flower".to_string(), "flow".to_string(), "flight".to_string()];
        assert_eq!(longest_common_prefix(strs), "fl".to_string());
    }

    #[test]
    fn test_example2() {
        let strs = vec!["dog".to_string(), "racecar".to_string(), "car".to_string()];
        assert_eq!(longest_common_prefix(strs), "".to_string());
    }
}
